File: CreditCard.c
#include <stdio.h>
#include <string.h>

int luhnCheck(char number[]) {
    int n = strlen(number);
    int sum = 0;
    int isSecond = 0;  // start from rightmost digit

    for (int i = n - 1; i >= 0; i--) {
        int digit = number[i] - '0';  // convert char to int

        if (isSecond) {
            digit = digit * 2;
            if (digit > 9)
                digit = digit - 9;
        }

        sum += digit;
        isSecond = !isSecond;  // flip 0 <-> 1
    }

    return (sum % 10 == 0);
}

int main() {
    char number[50];

    printf("Enter a number: ");
    scanf("%s", number);

    if (luhnCheck(number))
        printf("Valid \n");
    else
        printf("Invalid \n");

    return 0;
}

File: CreditCard.py
def luhn_check(number: str) -> bool:
    n = len(number)
    total = 0
    is_second = False  

    for i in range(n - 1, -1, -1):  
        digit = int(number[i])  

        if is_second:
            digit *= 2
            if digit > 9:
                digit -= 9

        total += digit
        is_second = not is_second  # flip True <-> False

    return total % 10 == 0


def main():
    number = input("Enter a number: ").strip()
    if luhn_check(number):
        print("Valid")
    else:
        print("Invalid")


if __name__ == "__main__":
    main()
